lib/apytypes/_array_functions.py

```diff
@@ -411,9 +411,478 @@ def expand_dims(a, axis):
+        The initialized array with zeros.
+    """
+    try:
+        zeros = a.zeros
+    except AttributeError:
+        raise TypeError(f"Cannot make zeros array of type {type(a)}")
+
+    from apytypes import APyFixedArray, APyFloatArray
+
+    if isinstance(a, APyFixedArray):
+        int_bits = int_bits if int_bits is not None else a.int_bits
+        frac_bits = frac_bits if int_bits is not None else a.frac_bits
+        return zeros(shape=a.shape, int_bits=int_bits, frac_bits=frac_bits)
+
+    if isinstance(a, APyFloatArray):
+        exp_bits = exp_bits if exp_bits is not None else a.exp_bits
+        mantissa_bits = mantissa_bits if mantissa_bits is not None else a.man_bits
+        return zeros(shape=a.shape, exp_bits=exp_bits, man_bits=mantissa_bits)
+
+    raise ValueError("Only 'fixed' and 'float' array_types are defined")
+
+
+def ones_like(a, int_bits=None, frac_bits=None, exp_bits=None, mantissa_bits=None):
+    """
+    Return an array of ones with the same shape and type as a given array.
+    Defaults to `a` wordlength
+
+    Parameters
+    ----------
+    a : :class:`APyFloatArray` or :class:`APyFixedArray`
+        The shape and data-type of a define these same attributes of the returned array.
+    int_bits : int, optional
+        Number of integer bits.
+    frac_bits : int, optional
+        Number of fractional bits.
+    exp_bits : int, optional
+        Number of exponential bits.
+    mantissa_bits : int, optional
+        Number of mantissa bits.
+
+    Returns
+    -------
+    result : :class:`APyFloatArray` or :class:`APyFixedArray`
+        The array initialized filled with ones.
+    """
+    try:
+        ones = a.ones
+    except AttributeError:
+        raise TypeError(f"Cannot make ones array of type {type(a)}")
+
+    from apytypes import APyFixedArray, APyFloatArray
+
+    if isinstance(a, APyFixedArray):
+        int_bits = int_bits if int_bits is not None else a.int_bits
+        frac_bits = frac_bits if frac_bits is not None else a.frac_bits
+        return ones(shape=a.shape, int_bits=int_bits, frac_bits=frac_bits)
+
+    if isinstance(a, APyFloatArray):
+        exp_bits = exp_bits if exp_bits is not None else a.exp_bits
+        mantissa_bits = mantissa_bits if mantissa_bits is not None else a.man_bits
+        return ones(shape=a.shape, exp_bits=exp_bits, man_bits=mantissa_bits)
+
+    raise ValueError("Only 'fixed' and 'float' array_types are defined")
+
+
+def full_like(
+    a,
+    fill_value,
+    int_bits=None,
+    frac_bits=None,
+    exp_bits=None,
+    mantissa_bits=None,
+):
+    """
+    Return a full array with the same shape and type as a given array.
+
+    If fill_value is an int or float, you must specify the word lengths (int_bits, frac_bits or exp_bits, mantissa_bits).
+    If fill_value is an APyFloat or APyFixed, the array will use the provided word lengths if specified.
+    If no word lengths are specified, the resulting array will inherit the word lengths from fill_value.
```

Comment:
If no word length are specified and `fill_value` is an `int` or `float` then I think it makes sense to inherit the word lengths of the reference array.